src/transport/capable_conn.rs

```diff
@@ -0,0 +1,84 @@
+use super::{CapableConn, QuicMuxedStream, RESET_ERR_CODE, Transport};
+use crate::{PeerId, PublicKey, multiaddr::{MultiaddrExt, Multiaddr}};
+
+use anyhow::Error;
+use async_trait::async_trait;
+use futures::{lock::Mutex, stream::StreamExt};
+use quinn::{Connection, ConnectionError, IncomingBiStreams};
+
+use std::sync::{Arc, atomic::{AtomicBool, Ordering}};
+
+#[derive(Clone)]
+pub struct QuicConn<Transport> {
+    conn: Connection,
+    bi_streams: Arc<Mutex<IncomingBiStreams>>,
```

Comment:
IncomingBiStreams makers Send + Sync，why add the `Mutex`  in here?

Reply:
Yes, but ```Stream``` trait from futures require ```mut```. ```Arc::get_mut``` require no other
```Arc``` and ```Weak``` pointers to same ```bi_streams```.  Most of time, we will only accept
new streams in one spawned task, i think it's ok here.